firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Homepage Rebuild/HomepageDiffableDataSourceTests.swift

```diff
@@ -29,17 +29,38 @@ final class HomepageDiffableDataSourceTests: XCTestCase {
     }
 
     // MARK: - applyInitialSnapshot
-    func test_applyInitialSnapshot_hasCorrectData() throws {
+    func test_updateSnapshot_hasCorrectData() throws {
         let dataSource = try XCTUnwrap(diffableDataSource)
 
-        dataSource.applyInitialSnapshot(state: HomepageState(windowUUID: .XCTestDefaultUUID))
+        dataSource.updateSnapshot(state: HomepageState(windowUUID: .XCTestDefaultUUID))
 
         let snapshot = dataSource.snapshot()
         XCTAssertEqual(snapshot.numberOfSections, 4)
-        XCTAssertEqual(snapshot.sectionIdentifiers, [.header, .topSites, .pocket, .customizeHomepage])
+        XCTAssertEqual(snapshot.sectionIdentifiers, [.header, .topSites, .pocket(nil), .customizeHomepage])
 
         XCTAssertEqual(snapshot.itemIdentifiers(inSection: .header).count, 1)
-        XCTAssertEqual(snapshot.itemIdentifiers(inSection: .pocket).count, 1)
+        XCTAssertEqual(snapshot.itemIdentifiers(inSection: .pocket(nil)).count, 1)
         XCTAssertEqual(snapshot.itemIdentifiers(inSection: .customizeHomepage).count, 1)
     }
+
+// what is actually the best way to test this? Should our state values be public vars? Probs not.
```

Comment:
Did you want feedback on this? Might be best to just book a meeting with someone on the team who's invested in Testing if you want test feedback. Maybe @cyndichin when she's back or @ih-codes

Reply:
Could also be a good iOS weekly Q and A discussion or Slack post

Reply:
lol whoops that was for me! I added the test and forgot to remove this comment. I will clean this up!